internal/services/netapp/netapp_volume_helper.go

```diff
@@ -273,6 +273,62 @@ func expandNetAppVolumeDataProtectionSnapshotPolicyPatch(input []interface{}) *v
 	}
 }
 
+func expandNetAppVolumeDataProtectionBackupPolicy(input []interface{}) *volumes.VolumePropertiesDataProtection {
+	if len(input) == 0 {
```

Comment:
```suggestion
	if len(input) == 0 || input[0] == nil {
```
We need to add a nil check here before casting to a map